activerecord/lib/active_record/connection_adapters/pool_config.rb

```diff
@@ -41,14 +40,19 @@ def server_version(connection)
         @server_version || synchronize { @server_version ||= connection.get_database_version }
       end
 
-      def connection_name
-        if connection_class.primary_class?
-          "ActiveRecord::Base"
+      def connection_class=(connection_class)
+        case connection_class
+        when ConnectionHandler::StringConnectionName
+          @connection_class = connection_class
         else
-          connection_class.name
+          @connection_class = ConnectionHandler::StringConnectionName.new(connection_class.name, connection_class.primary_class?)
```

Comment:
There's just one minor thing that bother me a bit, it is that now `@connection_class` is always a `StringConnectionName` instance, hence the name makes little sense.


Feels like we should refactor further. AFAIK this is all private API.

Reply:
I agree. I'm trying to craft a test that reproduces the issue, and then I will refactor and clean up. I appreciate the feedback!